utils.sh

```diff
@@ -148,7 +148,7 @@ function cleanup() {
   rm -rf ${tmpdir}
 
   local imageId=$(docker images -q ${repo}:${after})
-  if [ -n ${imageId} ]; then
+  if [ -n "${imageId}" ]; then
```

Comment:
bash was getting past this check in all cases, leading to `docker rmi -f` calls with no `imageId`. Tests above confirmed this and now pass with fix in place.